app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -325,23 +363,111 @@ export default Controller.extend(ModalFunctionality, {
     return this.model.links.filter((link) => !link._destroy);
   },
 
+  get activeSecondaryLinks() {
+    return this.model.secondaryLinks?.filter((link) => !link._destroy);
+  },
+
   get header() {
     return this.model.id
       ? "sidebar.sections.custom.edit"
       : "sidebar.sections.custom.add";
   },
 
+  @bind
+  reorder(linkIdFrom, linkIdTo, above) {
+    if (linkIdFrom === linkIdTo) {
+      return;
+    }
+    let fromLink = this.model.links.find(
+      (link) => link.objectId === linkIdFrom
+    );
+    if (!fromLink) {
+      fromLink = this.model.secondaryLinks.find(
+        (link) => link.objectId === linkIdFrom
+      );
+    }
+    this.model.links.removeObject(fromLink);
+    this.model.secondaryLinks?.removeObject(fromLink);
+
+    let toPosition = this.model.links.findIndex(
+      (link) => link.objectId === linkIdTo
+    );
+    if (toPosition === -1) {
+      fromLink.segment = "secondary";
+      toPosition = this.model.secondaryLinks.findIndex(
+        (link) => link.objectId === linkIdTo
+      );
+      if (above) {
+        this.model.secondaryLinks.insertAt(toPosition, fromLink);
+      } else {
+        this.model.secondaryLinks.insertAt(toPosition + 1, fromLink);
+      }
+    } else {
+      fromLink.segment = "primary";
+      if (above) {
+        this.model.links.insertAt(toPosition, fromLink);
+      } else {
+        this.model.links.insertAt(toPosition + 1, fromLink);
+      }
+    }
+  },
+
+  get canDelete() {
+    return this.model.id && !this.model.sectionType;
+  },
+
+  @bind
+  deleteLink(link) {
+    if (link.id) {
+      link._destroy = "1";
+    } else {
+      this.model.links.removeObject(link);
```

Comment:
I think we can check the return value here to see if an object has indeed been removed. If so, we can avoid removing the object from the secondary link. Also, wouldn't we be able to decide which array to remove the link from based on the segment property of the link?